cmd/erasure-object.go

```diff
@@ -412,7 +422,16 @@ func (er erasureObjects) getObjectFileInfo(ctx context.Context, bucket, object s
 func (er erasureObjects) getObjectInfo(ctx context.Context, bucket, object string, opts ObjectOptions) (objInfo ObjectInfo, err error) {
 	fi, _, _, err := er.getObjectFileInfo(ctx, bucket, object, opts)
 	if err != nil {
+		// proxy HEAD to replication target if active-active replication configured on bucket
+		if errors.Is(err, errFileNotFound) ||
+			errors.Is(err, errFileVersionNotFound) {
+			oi, proxy, err := proxyHeadToReplicationTarget(ctx, bucket, object, opts)
```

Comment:
We cannot allow this level of fallback in the hot path such as GetObject() we should not even enter the function without checking a server config setting or client request.

Reply:
same reasoning as above.